src/users/amoc_users_worker_sup.erl

```diff
@@ -0,0 +1,138 @@
+%% @private
+%% @copyright 2024 Erlang Solutions Ltd.
+%% @doc Supervisor-like gen_server with some tracking responsibilities over the users
+%%
+%% We want to keep consistent track of all users running globally by running special code upon a
+%% children's death. Standard solutions don't cut it because:
+%%  - A supervisor doesn't expose callbacks on user termination
+%%  - Implementing code on the user process before it dies risks inconsistencies if it is killed
+%%  More improvements that could be made would be to distribute the supervision tree like ranch did,
+%%  see https://stressgrid.com/blog/100k_cps_with_elixir/
+%% @end
+-module(amoc_users_worker_sup).
+
+-behaviour(gen_server).
+
+%% gen_server callbacks
+-export([start_link/1]).
+-export([init/1, handle_call/3, handle_cast/2, handle_info/2, terminate/2]).
+
+-record(state, {
+          index :: non_neg_integer(),
+          tid :: ets:tid(),
+          tasks = #{} :: #{reference() => pid()}
+         }).
+-type state() :: #state{}.
+
+-define(SHUTDOWN_TIMEOUT, 2000). %% 2 seconds
+
+%% @private
+-spec start_link(non_neg_integer()) -> {ok, pid()}.
+start_link(N) ->
+    gen_server:start_link(?MODULE, N, []).
+
+%% @private
+-spec init(non_neg_integer()) -> {ok, term()}.
+init(N) ->
+    process_flag(trap_exit, true),
+    Tid = ets:new(?MODULE,  [ordered_set, private]),
```

Comment:
any particular reason to make it private? why not protected and not named?
these settings make troubleshooting much harder.